### PR TITLE
Hengle/auto resources

### PR DESCRIPTION
**Pull request by jianghengle**
Some system admins in our center are interested in using crecto admin as a UI for them the manipulate the database. But they want to be able generate the code automatically given a database or repo. So I made a function reading the Mysql information_schema database to get all tables and columns and then generate the admin code automatically.
- Auto resource function
- Some primary field has no auto_increment function. When create the record, allow the user to set this primary field if he specify the primary field in the form attributes method
- Only select attributes defined in the model from the database. Because I run into a problem that the database adapter cannot retrieve some fields with weird types, even I did not define it in the model. So I only select defined fields to work around the bug.
- Added resource index in the dashboard page.

### CODE REVIEW

File: src/CrectoAdmin/auto_resource.cr [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    String.build do |s|
+      s << "Kemal::Session.config do |config|\n"
+      s << "  config.secret = \"my super secret\"\n"
+      s << "end\n"
+      s << "Kemal.run\n\n"
+    end
+  end
+end
+
+def self.auto_script(repo)
```
Comment (fridgerator) [inline src/CrectoAdmin/auto_resource.cr]: Do you know if this will also work for postgres / sqlite ?

File: src/CrectoAdmin/auto_resource.cr [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      @columns.each do |column|
+        next if column == primary_key_column
+        next if column.column_type.nil?
+        return true
+      end
+      false
+    end
+  end
+
+  def self.build_header(uri)
```
Comment (fridgerator) [inline src/CrectoAdmin/auto_resource.cr]: Building strings is fine, just so you are aware ECR templates can also be used to generate files: https://crystal-lang.org/api/0.26.0/ECR.html

Comment (jianghengle) [reply src/CrectoAdmin/auto_resource.cr]: Oh, I did not realize that. Yeah, that will be much better to generate crystal code. I will definitely change to that.

File: src/CrectoAdmin/auto_resource.cr (hunk 1 above)
Comment (jianghengle) [reply src/CrectoAdmin/auto_resource.cr]: No, I do not think so. I think it only works for mysql. I did not put this in readme. Inside of the example, I stated that it only works for mysql. Even for mysql, the generated admin may not work for all tables, and needs to be reviewed. I view it just a helper for users that do not want to type code at start. And the our admins here seems really want this kind of helper.

**Pull request merged**